cordova-lib/src/hooks/HooksRunner.js

```diff
@@ -178,7 +178,9 @@ function runScriptViaModuleLoader(script, context) {
     }
     var scriptFn = require(script.fullPath);
     context.scriptLocation = script.fullPath;
-    context.opts.plugin = script.plugin;
+    if(script.plugin) {
+        context.opts.plugin = script.plugin;
```

Comment:
Yes and no. cordova-lib/src/plugman/install.js already populates `opts.plugin` before it is passed to `HooksRunner`.
What currently happens in this code snippet is that the field is overridden with either information on the plugin which provides the script (i.e. the on in which `plugin.xml` it is defined) or `undefined` if the script is part of the project (i.e. defined in config.xml) because `script.plugin` is `undefined`.
My change just prevents the override in the later case. The first case should behave as before so it doesn't break existing scripts.